pandas/tests/extension/base/groupby.py

```diff
@@ -19,6 +19,7 @@ def test_grouping_grouper(self, data_for_grouping):
         tm.assert_numpy_array_equal(gr1.grouper, df.A.values)
         tm.assert_extension_array_equal(gr2.grouper, data_for_grouping)
 
+    @pytest.mark.skip(reason="logic change to stop coercing dtypes on set_index()")
```

Comment:
So you are skipping these because the existing tests are incompatible with the new logic right? 

If that's correct then we should actually update the tests and not just skip

Reply:
Yes, I will refactor them. From what I could see looking through tests, it seemed to me like a lot of them are skipped rather than refactored. 

Would best practice be to try and always refactor? Thanks for your feedback

Reply:
This PR is rarer than others as it intentionally is breaking existing behavior; in such case yes would want to update existing tests and not just skip them

Also needs a little more communication in whatsnew around backwards compatibility but can leave that until later; I think would be helpful to see tests updated for reviewers to have a better understanding of scope of changes this would cause

Reply:
Okay, that make sense. I'll work on getting these tests updated so it is clearer what the scope of the change is. In general, these few I skipped were asserting two df's to be equal, which is true *except* of course for the index of each df were no longer equal, thus they failed.

Reply:
yes, you need to actually change the results